### ISSUE TITLE
Crash with Deferred data aborted

### ISSUE THREAD

**Issue by lukasa1993** [forge42dev/Remix-Dev-Tools#96]
remix vite expressjs 

way to reproduce is to fast navigate from and away of two pages with deferred values

sample: 
https://codesandbox.io/p/devbox/vite-rdt-test-l36c2f

**Comment by AlemTuzlak** [comment]
@lukasa1993 can you try this with version 3.7.4

**Comment by lukasa1993** [comment]
> @lukasa1993 can you try this with version 3.7.4

still crashes :(

**Comment by AlemTuzlak** [comment]
@lukasa1993 i think i tracked down the cause, hopefully fixed it. Can you try now?

**Comment by lukasa1993** [comment]
> @lukasa1993 i think i tracked down the cause, hopefully fixed it. Can you try now? With v4

still crashed :( 


```2024-03-11T09:59:17.324Z ::1 - GET /item/3da8072a-e3ad-4d47-866c-c96b5c8428ad?_data=routes%2Fdeal HTTP/1.1 200 - - 136.642 ms
/node_modules/@remix-run/router/dist/router.cjs.js:1324
    let onAbort = () => reject(new AbortedDeferredError("Deferred data aborted"));
                               ^

AbortedDeferredError: Deferred data aborted
    at EventTarget.onAbort (/node_modules/@remix-run/router/dist/router.cjs.js:1324:32)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:786:20)
    at EventTarget.dispatchEvent (node:internal/event_target:721:26)
    at abortSignal (node:internal/abort_controller:369:10)
    at AbortController.abort (node:internal/abort_controller:403:5)
    at DeferredData.cancel (/node_modules/@remix-run/router/dist/router.cjs.js:1403:21)
    at EventTarget.onAbort (/node_modules/@remix-run/router/dist/router.cjs.js:1410:32)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:786:20)
    at EventTarget.dispatchEvent (node:internal/event_target:721:26)
    at abortSignal (node:internal/abort_controller:369:10)
    at AbortController.abort (node:internal/abort_controller:403:5)
    at ServerResponse.<anonymous> (/node_modules/@remix-run/express/dist/server.js:79:36)
    at ServerResponse.emit (node:events:517:28)
    at emitCloseNT (node:_http_server:1012:10)
    at Socket.onServerResponseClose (node:_http_server:278:5)
    at Socket.emit (node:events:529:35)
    at TCP.<anonymous> (node:net:350:12)
```

**Comment by AlemTuzlak** [comment]
@lukasa1993 I actually think I tracked down the issue and fixed it with v4.2.1
